.github/workflows/android.yml

```diff
@@ -24,6 +26,11 @@ jobs:
           distribution: 'temurin'
           cache: gradle
 
+      - name: Decode and write google service json
```

Comment:
In general, we can reuse steps by defining a yaml anchor before, e.g before jobs:

```yml
firebase-setup: &firebase-setup
  - name: Decode and write google service json
    env:
      GOOGLE_SERVICES_JSON: ${{ secrets.GOOGLE_SERVICES_JSON }}
    run: echo $GOOGLE_SERVICES_JSON | base64 --decode >./app/google-services.json
```

and reuse it here: `- <<: *firebase-setup` and in other jobs.

Also, we need to inject the json only for jobs that require a project build (`gradlew build`). Do dokka, lint and test depend on build? Maybe test yes, but not sure whether it is needed for the others

Reply:
That's a really cool idea; I have not used that before.
Unfortunately, according to [this issue actions/runner/#1182](https://github.com/actions/runner/issues/1182) this syntax doesn't seem to be supported in the GitHub Action YAML.

Reply:
I had also assumed that only build and test needed the google json, but the github action only went through when I added the google-services.json at every single step